File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/security/AccountGroupContributor.java
package net.nemerosa.ontrack.model.security;

import javax.validation.constraints.NotNull;
import java.util.Collection;

/**
 * FIXME Contributes a list of groups to an account.
 */
public interface AccountGroupContributor {

    /**
     * Collects the list of groups for this account
     *
     * @param authenticatedAccount Account with authentication information to collect groups for
     * @return List of groups (can be empty but not null)
     */
    // Collection<AccountGroup> collectGroups(@NotNull AuthenticatedAccount authenticatedAccount);

}

File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/security/AccountGroupContributor.kt
package net.nemerosa.ontrack.model.security

import org.springframework.stereotype.Component

/**
 * Contributes a list of groups to an account.
 */
interface AccountGroupContributor {

    /**
     * Collects the list of groups for this account
     *
     * @param account Account
     * @return List of groups
     */
    fun collectGroups(account: Account): Collection<AccountGroup>

}

/**
 * NOP contributor
 */
@Component
class NOPAccountGroupContributor : AccountGroupContributor {
    override fun collectGroups(account: Account): Collection<AccountGroup> = emptySet()
}
